browser/scripts/ui-core.js

```diff
@@ -68,6 +97,153 @@ VizorUI.prototype._init = function(e2) {	// called by .init() in ui.js
 };
 
 
+VizorUI.prototype.updateState = function(e) {
+	this.state._update({
+		'chat':		this.dom.chatWindow,
+		'presets':	this.dom.presetsLib,
+		'assets':	this.dom.assetsLib
+	});
+	this.emit(uiEvents.state_updated, this.state._getCopy());
+	return true;
+}
+
+VizorUI.prototype.setupState = function() {
+	var ui = this;
+
+	this.state._update = function(domPanelStates) {	/* @var domPanelStates {key: $domElement, ...} */
+		if (!ui._initialised) return false;
+		// this = ui.state
+		this.context.availWidth = window.screen.availWidth;
+		this.context.availHeight = window.screen.availHeight;
+		var ps = this.panelStates;					// ref
+		if (typeof domPanelStates === 'object') {
+			var key;
+			for (key in domPanelStates) {
+				if (domPanelStates.hasOwnProperty(key)) {
+					ps[key] = VizorUI.getDomPanelState(domPanelStates[key]);
+				}
+			}
+		}
+		return this;
+	};
+
+	this.state._apply = function(newState) {
+		if ((!ui._initialised) || (typeof newState !== 'object')) return false;
```

Comment:
this should be an error. if newState is not an object or this is called without having the UI initialised, now this just fails quietly, which could lead to bugs that are really hard to track because nothing returns an error